File: backend/internal/model/grafana_instance.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

// GrafanaInstance Grafana 实例连接信息（平台统一管理）。
// source: platform（Zone 自动部署）/ external（管理员手动登记）。
type GrafanaInstance struct {
	ID            uuid.UUID      `json:"id" gorm:"type:uuid;primaryKey"`
	Name          string         `json:"name" gorm:"type:varchar(255);not null"`
	Source        string         `json:"source" gorm:"type:varchar(20);default:external;index"` // platform / external
	ZoneID        *uuid.UUID     `json:"zone_id" gorm:"type:uuid;index"`
	URL           string         `json:"url" gorm:"type:varchar(500)"`
	AdminUser     string         `json:"admin_user" gorm:"type:varchar(100)"`
	AdminPassword string         `json:"-" gorm:"type:varchar(255)"`
	AdminTokenEnc string         `json:"-" gorm:"type:text"`
	Status        string         `json:"status" gorm:"type:varchar(20);default:active"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
	DeletedAt     gorm.DeletedAt `json:"-" gorm:"index"`
}

// TableName 表名。
func (GrafanaInstance) TableName() string {
	return "ops_grafana_instances"
}

// BeforeCreate 生成主键。
func (g *GrafanaInstance) BeforeCreate(tx *gorm.DB) error {
	if g.ID == uuid.Nil {
		g.ID = uuid.New()
	}
	return nil
}

File: frontend/src/api/grafanaInstance.ts
import api from './index';
import type { AxiosRequestConfig } from 'axios';
import type { ApiResponse, PaginatedResponse, PaginationParams } from '../types/api';

export interface GrafanaInstance {
  id: string;
  name: string;
  source: 'platform' | 'external';
  zone_id: string | null;
  url: string;
  admin_user: string;
  status: string;
  created_at: string;
  updated_at: string;
}

export const grafanaInstanceAPI = {
  list: (params?: PaginationParams & { source?: string; zone_id?: string }, config?: AxiosRequestConfig) =>
    api.get<ApiResponse<PaginatedResponse<GrafanaInstance>>>('/grafana/instances', { ...config, params }),

  get: (id: string, config?: AxiosRequestConfig) =>
    api.get<ApiResponse<GrafanaInstance>>(`/grafana/instances/${id}`, config),

  create: (data: {
    name: string;
    source: 'platform' | 'external';
    zone_id?: string;
    url: string;
    admin_user?: string;
    admin_password?: string;
    admin_token?: string;
  }) => api.post<ApiResponse<GrafanaInstance>>('/grafana/instances', data),

  update: (id: string, data: Partial<GrafanaInstance> & { admin_password?: string; admin_token?: string }) =>
    api.put<ApiResponse<GrafanaInstance>>(`/grafana/instances/${id}`, data),

  delete: (id: string) =>
    api.delete<ApiResponse<null>>(`/grafana/instances/${id}`),

  login: (id: string, redirect?: string) =>
    api.post<ApiResponse<{ proxyUrl: string }>>(`/grafana/instances/${id}/login`, null, {
      params: redirect ? { redirect } : undefined,
    }),
};
